test/unit_tests/qir2qasm_trans/test_end_to_end.py

```diff
@@ -0,0 +1,270 @@
+        source_filename = "bell"
+
+        %Qubit = type opaque
+        %Result = type opaque
+
+        define void @main() #0 {
+        entry:
+        call void @__quantum__qis__h__body(%Qubit* null)
+        call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+        call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+        call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* inttoptr (i64 1 to %Result*))
+        ret void
+        }
+
+        declare void @__quantum__qis__h__body(%Qubit*)
+
+        declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+        declare void @__quantum__qis__mz__body(%Qubit*, %Result* writeonly) #1
+
+        attributes #0 = { "entry_point" "output_labeling_schema" "qir_profiles"="custom" "required_num_qubits"="2" "required_num_results"="2" }
+        attributes #1 = { "irreversible" }
+
+        !llvm.module.flags = !{!0, !1, !2, !3}
+
+        !0 = !{i32 1, !"qir_major_version", i32 1}
+        !1 = !{i32 7, !"qir_minor_version", i32 0}
+        !2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+        !3 = !{i32 1, !"dynamic_result_management", i1 false}
+    """
+
+    # Expected QASM output
+    expected_qasm = """
+        OPENQASM 3.0;
+        include "stdgates.inc";
+        qubit[2] Qubits;
+        bit[2] Results;
+        h Qubits[0];
+        cx Qubits[0], Qubits[1];
+        Results[0] = measure Qubits[0];
+        Results[1] = measure Qubits[1];
+    """
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        qir_path = Path(tmp_dir) / "bell.ll"
+        qir_path.write_text(textwrap.dedent(qir_content), encoding="utf-8")
+
+        # Load the QIR module
+        module = load(str(qir_path))
+
+        # Convert to QASM using the Exporter
+        exporter = Exporter()
+        qasm_output = exporter.dumps(module)
+
+        # Validate the complete output
+        assert qasm_output.strip() == textwrap.dedent(expected_qasm).strip()
+
+
+def test_bell_pyqir_to_qasm():
+    """Test end-to-end conversion of a Bell pair QIR program to QASM."""
+    bell = pyqir.SimpleModule("bell", num_qubits=2, num_results=2)
+    qis = pyqir.BasicQisBuilder(bell.builder)
+
+    # Add instructions to the module to create a Bell pair and measure both qubits.
+    qis.h(bell.qubits[0])
+    qis.cx(bell.qubits[0], bell.qubits[1])
+    qis.mz(bell.qubits[0], bell.results[0])
+    qis.mz(bell.qubits[1], bell.results[1])
+
+    # Sample QIR content for a Bell pair circuit
+    # Expected QASM output
+    expected_qasm = """
+        OPENQASM 3.0;
+        include "stdgates.inc";
+        qubit[2] Qubits;
+        bit[2] Results;
+        h Qubits[0];
+        cx Qubits[0], Qubits[1];
+        Results[0] = measure Qubits[0];
+        Results[1] = measure Qubits[1];
+    """
+
+    # Create a temporary QIR file
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".ll", delete=True) as temp_file:
```

Comment:
The files is opened twice, once during `NamedTemporaryFile()` and once during `load()`. This is okay in Linux but cause Windows to fail. This is one of the reasons github checks for Windows fail.

A solution is to replace `NamedTemporaryFile` with `TemporaryDirectory`. I tested in "test_bell_qir_to_qasm" and it appears to be working. Can you make the similar change for other tests so that checks on Windows pass?